packages/config/src/Config.ts

```diff
@@ -434,16 +461,17 @@ export class Config {
      * @param schema The URI of JSON schema, or a schema object to use
      */
     public setSchema(schema: string | object) {
+        const layer = this.layerActive();
         const schemaUri = (typeof schema === "string") ? schema : `./${this.schemaName}`;
         const schemaObj = (typeof schema !== "string") ? schema : null;
 
-        const layer = this.layerActive();
-        delete layer.properties.$schema;
-        layer.properties = { $schema: schemaUri, ...layer.properties };
+        if (layer.properties.$schema == null) {
+            layer.properties = JSONC.parse(JSONC.stringify({ ...{ $schema: schemaUri }, ...layer.properties }, null, ConfigConstants.INDENT));
+        }
 
-        if (schemaObj != null) {
-            const filePath = node_path.resolve(node_path.dirname(layer.path), schemaUri);
-            fs.writeFileSync(filePath, JSONC.stringify(schemaObj, null, ConfigConstants.INDENT));
+        const schemaInfo = this.getSchemaInfo();
+        if (schemaObj != null && (schemaInfo.local || schemaInfo.original.startsWith("./"))) {
```

Comment:
I'm particularly curious about this, because the import handler uses different logic for determining whether a config is local:
`const isConfigLocal = fs.existsSync(configFilePath) || path.isAbsolute(params.arguments.location);`

In theory, the `path.isAbsolute` check should accurately determine whether or not a file is local, even if it doesn't exist.

Reply:
> Is it because the schema may be local, even though it doesn't exist on disk yet?

That's was the reason.

> the `path.isAbsolute` check should accurately determine whether or not an absolute URL/path is local, even if it doesn't exist

That's right, however relative paths that do not exists will also return `local == false`
Then we could just resolve it prior to checking it, right? ✅ 
But what happens to URLs?
`path.isAbsolute(path.resolve("http:/localhost:12345/zowe.schema.json"))`
`path.isAbsolute("/root/gh/zowe/imperative/http:/localhost:12345/zowe.schema.json")`
Results: `true`

TL;DR: No good way to workaround it other than
```
let isUrl = true; 
try { new URL(params.arguments.location) } catch (_) { isUrl = false; }
```

Reply:
Handled URLs as mentioned in the comment above.
Q: Should the `isUrl` be a utility function on Imperative?